glslc/src/file_compiler.cc

```diff
@@ -168,8 +197,36 @@ bool FileCompiler::EmitCompiledResult(
     }
   }
 
-  // Write compilation output to output file.
-  out->write(compilation_output.data(), compilation_output.size());
+  // Write compilation output to output file. If an output format for SPIR-V
+  // binary code is specified, it is handled here.
+  switch (binary_emission_format_) {
```

Comment:
I find it weird to asert the output format *inside* the switch.  We should just put the entire switch into an if(output format == binary) clause.